src/client/dfs/dfs.c

```diff
@@ -3575,6 +3586,16 @@ dfs_update_parent(dfs_obj_t *obj, dfs_obj_t *src_obj, const char *name)
 	return 0;
 }
 
+/* Update a object to a new parent, taking the parent directly */
+void
+dfs_update_parentfd(dfs_obj_t *obj, dfs_obj_t *src_obj, const char *name)
+{
+	oid_cp(&obj->parent_oid, src_obj->oid);
```

Comment:
so here you are just changing the parent of the open handle.
but you never insert the handle in the new parent or remove it from the older one?
you probably need this just for dfuse, but the comment in the header need to be more explicit as what this is doing to not give wrong impression

Reply:
No, this is similar to the existing dfs_update_parent() function which as you say works on a struct not a daos object although the type is confusing.  The existing function takes two open handles and sets the parent of one to be the same as the parent on the other, but dfuse also needs to be able to set the parent of an open handle to a open handle directly.